src/main/java/htsjdk/samtools/reference/FastaReferenceWriter.java

```diff
@@ -0,0 +1,674 @@
+     */
+    private static final char LINE_SEPARATOR_CHR = '\n';
+
+    /**
+     * Character used to separate the fields in a index file line.
+     */
+    private static final char INDEX_FIELD_SEPARATOR_CHR = '\t';
+
+    /**
+     * Convenient cached {@code byte[]} representation of the line separator.
+     */
+    private static final byte[] LINE_SEPARATOR = String.valueOf(LINE_SEPARATOR_CHR).getBytes(CHARSET);
+
+    /**
+     * Output stream to the main FASTA output.
+     * <p>
+     * We use it also to count the number of bytes so far outputted thus the offset included in
+     * the index file entry.
+     * </p>
+     */
+    private final CountingOutputStream fastaStream;
+
+    /**
+     * Writer for the index file.
+     */
+    private final Writer indexWriter;
+
+    /**
+     * FastaSequenceIndex for the index file.
+     */
+    private final FastaSequenceIndex index = new FastaSequenceIndex();
+
+    /**
+     * Output writer to the output dictionary.
+     */
+    private final Writer dictWriter;
+
+    /**
+     * Output codec for the dictionary.
+     */
+    private final SAMSequenceDictionaryCodec dictCodec;
+
+    /**
+     * Default number of bases per line to be applied unless one is
+     */
+    private final int defaultBasePerLine;
+
+    /**
+     * Records the sequences that have been already fully appended to this writer.
+     * <p>
+     * The key is the sequence name.
+     * </p>
+     * <p>
+     * The value is the sequence length in bases.
+     * </p>
+     */
+    private final Map<String, Long> sequenceNamesAndSizes = new LinkedHashMap<>();
+
+    /**
+     * Bases per line to be applied to the sequence that is been currently appended to the output.
+     */
+    private int currentBasesPerLine;
+
+    /**
+     * Holds the number of bases in the current output line.
+     */
+    private int currentLineBasesCount;
+
+    /**
+     * Holds the number of bases so far appended for the current sequence.
+     */
+    private long currentBasesCount;
+
+    /**
+     * Holds the FASTA output file offset for the current sequence.
+     */
+    private long currentSequenceOffset;
+
+    /**
+     * Holds the name of the sequence that is been appended currently.
+     */
+    private String currentSequenceName;
+
+    /**
+     * Flag indicating whether this writer has been already closed.
+     */
+    private boolean closed;
+
+    /**
+     * Creates a reference FASTA file writer (private...use the builder: {@link FastaReferenceWriterBuilder}.
+     * <p>
+     * You can specify a specific output stream to each file: the main fasta output, its index and its dictionary.
+     * </p>
+     *
+     * @param fastaOutput the output fasta file path.
+     * @param indexOutput the output stream to the index file, if requested, {@code null} if none should be generated.
+     * @param dictOutput  the output stream to the dictFile, if requested, {@code null} if none should be generated.
```

Comment:
Specify that both index/dict streams should not be compressed (the same in the builder).